test/hotspot/jtreg/compiler/integerArithmetic/DivisionByConstant.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.integerArithmetic;
+
+import compiler.lib.ir_framework.*;
+import java.util.Random;
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8282365
+ * @summary Test that Ideal transformations of division nodes provide correct
+ * result.
+ *
+ * @requires os.simpleArch=="x64"
```

Comment:
Can you move the `x64` requirement to the IR rule? This has just been made possible with [JDK-8280120](https://bugs.openjdk.org/browse/JDK-8280120)

Reply:
Same with your other tests. That way, we can at least make sure that the results are correct on all other platforms.

Reply:
Done